contracts/poap/src/msg.rs

```diff
@@ -0,0 +1,398 @@
+                minter: "".to_string(),
+                symbol: "".to_string(),
+            },
+            event_info: EventInfo {
+                creator: "".to_string(),
+                start_time: Timestamp::from_seconds(1),
+                end_time: Timestamp::from_seconds(2),
+                per_address_limit: 1,
+                base_poap_uri: "ipfs://domain.com".to_string(),
+                event_uri: "https://domain.com".to_string(),
+            },
+        };
+
+        assert_eq!(
+            ContractError::InvalidEventUri {},
+            msg.validate().unwrap_err()
+        );
+    }
+
+    #[test]
+    fn instantiate_with_invalid_base_poap_uri_error() {
+        // Invalid uri
+        let msg = InstantiateMsg {
+            admin: "".to_string(),
+            minter: "".to_string(),
+            cw721_code_id: 0u64.into(),
+            cw721_initiate_msg: Cw721InstantiateMsg {
+                name: "".to_string(),
+                minter: "".to_string(),
+                symbol: "".to_string(),
+            },
+            event_info: EventInfo {
+                creator: "".to_string(),
+                start_time: Timestamp::from_seconds(1),
+                end_time: Timestamp::from_seconds(2),
+                per_address_limit: 1,
+                base_poap_uri: "https://domain.com".to_string(),
+                event_uri: "ipfs://domain.com".to_string(),
+            },
+        };
+
+        assert_eq!(
+            ContractError::InvalidPoapUri {},
+            msg.validate().unwrap_err()
+        );
+
+        // Non ipfs uri
+        let msg = InstantiateMsg {
+            admin: "".to_string(),
+            minter: "".to_string(),
+            cw721_code_id: 0u64.into(),
+            cw721_initiate_msg: Cw721InstantiateMsg {
+                name: "".to_string(),
+                minter: "".to_string(),
+                symbol: "".to_string(),
+            },
+            event_info: EventInfo {
+                creator: "".to_string(),
+                start_time: Timestamp::from_seconds(1),
+                end_time: Timestamp::from_seconds(2),
+                per_address_limit: 1,
+                base_poap_uri: "https://domain.com".to_string(),
+                event_uri: "ipfs://domain.com".to_string(),
+            },
+        };
+
+        assert_eq!(
+            ContractError::InvalidPoapUri {},
+            msg.validate().unwrap_err()
+        );
+    }
+
+    #[test]
+    fn update_event_info_start_time_after_end_time_error() {
+        // Start time after end time
+        let start = Timestamp::from_seconds(2);
+        let end = Timestamp::from_seconds(1);
+        let msg = ExecuteMsg::UpdateEventInfo {
+            start_time: start.clone(),
+            end_time: end.clone(),
+        };
+
+        assert_eq!(
+            ContractError::StartTimeAfterEndTime { start, end },
+            msg.validate().unwrap_err()
+        );
+
+        // Start time eq end time
```

Comment:
As above, I think it can be splitted to `update_event_info_start_time_eq_end_time_error`